runtime/resolvers/resolvers_test.go

```diff
@@ -0,0 +1,158 @@
+	require.NoError(t, err)
+	var reg = regexp.MustCompile(`^(.*)_resolvers_test.yaml$`)
+	for _, e := range entries {
+		if reg.MatchString(e.Name()) {
+			t.Log("Running with", e.Name())
+			yamlFile, err := os.ReadFile(e.Name())
+			require.NoError(t, err)
+			var r Resolvers
+			err = yaml.Unmarshal(yamlFile, &r)
+			require.NoError(t, err)
+
+			files := make(map[string]string)
+			for name, node := range r.Project.Sources {
+				abs := filepath.Join("sources", name)
+				bytes, err := yaml.Marshal(&node)
+				require.NoError(t, err)
+				files[abs] = string(bytes)
+			}
+			for name, node := range r.Project.Models {
+				abs := filepath.Join("models", name)
+				var bytes []byte
+				bytes, err = yaml.Marshal(&node)
+				require.NoError(t, err)
+				files[abs] = string(bytes)
+			}
+			for name, node := range r.Project.Dashboards {
+				abs := filepath.Join("dashboards", name)
+				bytes, err := yaml.Marshal(&node)
+				require.NoError(t, err)
+				files[abs] = string(bytes)
+			}
+			for name, node := range r.Project.APIs {
+				abs := filepath.Join("apis", name)
+				bytes, err := yaml.Marshal(&node)
+				require.NoError(t, err)
+				files[abs] = string(bytes)
+			}
+
+			for connector, opts := range r.Connectors {
+				t.Log("Running with", connector)
+				if opts == nil {
+					opts = &testruntime.InstanceOptionsForResolvers{}
+				}
+				if opts.Files == nil {
+					opts.Files = map[string]string{"rill.yaml": ""}
+				}
+
+				switch connector {
+				case "druid":
+					opts.OLAPDriver = "druid"
+				case "clickhouse":
+					opts.OLAPDriver = "clickhouse"
+				}
+
+				maps.Copy(opts.Files, files)
+				rt, instanceID := testruntime.NewInstanceForResolvers(t, *opts)
+				for testName, test := range r.Tests {
+					t.Run(testName, func(t *testing.T) {
+						t.Log("======================")
+						t.Log("Running ", testName, "with", e.Name(), "and", connector)
+						testruntime.RequireParseErrors(t, rt, instanceID, nil)
+						api, err := rt.APIForName(context.Background(), instanceID, test.Resolver)
+						require.NoError(t, err)
+
+						o := test.Options
+						o.InstanceID = instanceID
+						o.Resolver = api.Spec.Resolver
+						o.ResolverProperties = api.Spec.ResolverProperties.AsMap()
+
+						res, err := rt.Resolve(context.Background(), &o)
+						require.NoError(t, err)
+						var rows []map[string]interface{}
+						b, err := res.MarshalJSON()
+						require.NoError(t, err)
+						require.NoError(t, json.Unmarshal(b, &rows), string(b))
+						if *update || true {
+							test.Result = rows
+							for _, m := range test.Result {
+								for k, v := range m {
+									node := yaml.Node{}
+									node.Kind = yaml.ScalarNode
+									switch val := v.(type) {
+									case float32:
+										node.Value = strconv.FormatFloat(float64(val), 'f', 2, 32)
+										m[k] = &node
+									case float64:
+										node.Value = strconv.FormatFloat(val, 'f', 2, 64)
+										m[k] = &node
+									}
+								}
+							}
```

Comment:
Instead of trying to force YAML to deserialize a value into a specific type (it is a dynamically typed format after all), I think it would be simpler to just serialize the desired result from the YAML to JSON and comparing that to the resolver result's JSON (Go's serialized JSON output is mostly deterministic IIRC).

Reply:
"serialize the desired result from the YAML to JSON" puzzled here